### ISSUE TITLE
Terraform tries to apply resource with network from default region

### ISSUE THREAD

**Issue by AntonTumasov** [terraform-providers/terraform-provider-google#1258]
### Terraform Version
Terraform v0.11.3
Provider.google v1.3.0

### Affected Resource(s)
Please list the resources as a list, for example:
- *google_compute_instance*
- *google_compute_subnetwork*
- *provider*

### Expected Behavior
I have subnetwork and instance resources with the same values and provider section in *variables.tf* with different value.

I expect instance being applied with overridden zone from instance, not from value in provider section

### Actual Behavior
terraform failed with 400 code while applying 

### Steps to Reproduce
1. Create *variables.tf* file with provider "google" and  some region value
2. Create `google_compute_instance` resource with the different value in zone section where you subnetwork lives.
3. `terraform apply`
4. There is an error about zone being not in the region you set in the provider section in *variables.tf*

**Comment by emilymye** [comment]
Hi @AntonTumasov, can you provide the config you are using and debug logs? I'm not sure there is any region that we get from the provider for google_compute_instance, but possibly a seperate resource requires the region. For subnetworks, if you are using a different region than the given zone, you need to provide that zone or it will use the provider default (see [region parameter docs](https://www.terraform.io/docs/providers/google/r/compute_subnetwork.html#region)].

**Comment by AntonTumasov** [comment]
variables.tf 
```provider "google" {
  credentials = "credentials"
  project     = "example-project"
  region      = "us-central1"
  version     = "1.3"
}
```
instances.tf
``` resource "google_compute_instance" "gce-test" {
  name         = "gce-test"
  machine_type = "g1-small"
  zone         = "us-west1-a"
  tags         = ["test"]

  boot_disk {
    initialize_params {
      image = "centos7"
    }
    auto_delete = false
  }

  network_interface {
    subnetwork = "my-network"
    address    = "1.2.3.4"
    access_config {
      // Ephemeral IP
    }
  }

  scheduling {
    on_host_maintenance = "MIGRATE"
    automatic_restart = true
  }
}
```

network.tf 

```resource "google_compute_subnetwork" "my-network" {
  name          = "my-network"
  ip_cidr_range = "${var.subnets["test"]}"
  network       = "${google_compute_network.prod-network.self_link}"
  region        = "us-west1"
  description   = "Network for testing"
} 
```
So, eventually I get 404 error like that
``` Error: Error applying plan:

1 error(s) occurred:

* google_compute_instance.gce-test: 1 error(s) occurred:

* google_compute_instance.gce-test: Error creating instance: googleapi: Error 404: The resource 'projects/example-project/regions/us-central1/subnetworks/my-network' was not found, notFound

**Comment by rosbo** [comment]
Hi,

This issue was fixed and released in 1.5.0 and I see that you are using 1.3.0. Here is a link to the PR fixing it: #938.

If you are not ready to upgrade to the latest version of the provider, you can, as a workaround, use the `self_link` to refer to the subnetwork.

```hcl
...
network_interface {
  ...
  subnetwork = ${google_compute_subnetwork.my-network.self_link}
  ...
}
...
```
